tensorflow/java/BUILD

```diff
@@ -68,8 +68,13 @@ filegroup(
     ],
 )
 
+# Build the gen tool as a library, as it will be linked to a core/ops binary
+# files before making it an executable.
```

Comment:
I think it is still safe to select explicitly the list of libraries that we want to link to when we generate the gen tool library, since some of them are more tricky to handle and have been excluded for now (e.g. `dataset_ops`).

And we still need the `java_op_gen_lib` to be compiled separately from the binary because it is linked with generator unit tests, like `source_writer_test`

But yes, I guess the `java_op_gen_tool` binary rule could be in BUILD instead and I'll do an overall review of the comments I left there.

Reply:
But we aren't linking specific libraries (and can handle exclusions with `ApiDef` files instead?) Anyway, seems like you're looking to clean this up so that's great. Thanks!